messages/src/msg_fields/protocols/pickup/decorators.rs

```diff
@@ -0,0 +1,59 @@
+use serde::{Deserialize, Serialize};
+use typed_builder::TypedBuilder;
+
+use crate::decorators::thread::Thread;
+
+#[derive(Clone, Debug, Deserialize, Serialize, Default, PartialEq, TypedBuilder)]
+pub struct PickupDecoratorsCommon {
+    #[builder(default, setter(strip_option))]
+    #[serde(rename = "~thread")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub thread: Option<Thread>,
+    #[builder(default, setter(strip_option))]
+    #[serde(rename = "~transport")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub transport: Option<Transport>,
+}
```

Comment:
The general approach is to have a `MyMessageDecorators` type for each message. While it duplicates the data structures sometimes, it's easier to reason with and modify if needed. 

Additionally, the rest of the crate is like that, and I think consistency is also valuable enough to warrant having individual decorators types for each message.